cache.go

```diff
@@ -10,17 +11,19 @@ type Cache struct {
 
 // NewLRUCache creates a new LRU Cache object with the capacity given.
 func NewLRUCache(capacity uint64) *Cache {
-	return NewNativeCache(C.rocksdb_cache_create_lru(C.size_t(capacity)))
+	cCache := C.rocksdb_cache_create_lru(C.size_t(capacity))
+	return newNativeCache(unsafe.Pointer(cCache))
 }
 
 // NewLRUCacheWithOptions creates a new LRU Cache from options.
 func NewLRUCacheWithOptions(opt *LRUCacheOptions) *Cache {
-	return NewNativeCache(C.rocksdb_cache_create_lru_opts(opt.c))
+	cCache := C.rocksdb_cache_create_lru_opts(opt.c)
+	return newNativeCache(unsafe.Pointer(cCache))
 }
 
 // NewNativeCache creates a Cache object.
-func NewNativeCache(c *C.rocksdb_cache_t) *Cache {
-	return &Cache{c}
+func newNativeCache(c unsafe.Pointer) *Cache {
+	return &Cache{c: (*C.rocksdb_cache_t)(c)}
```

Comment:
I prefer using `*C.rocksdb_...` here to leverage **compiler type-checker**. 

I don't want to pass in generic `unsafe.Pointer` here because it will disable type-check -> error prone

HDYT @muthukrishnan24

Reply:
@linxGnu changed, please check